raven/processes/wps_nalcms_zonal_stats_raster.py

```diff
@@ -0,0 +1,284 @@
+    "Wetland": [14],
+    "Crops": [15],
+    "Urban": [17],
+    "Water": [18],
+    "SnowIce": [19],
+}
+SIMPLE_CATEGORIES = {i: cat for (cat, ids) in simplified.items() for i in ids}
+
+SUMMARY_ZONAL_STATS = ["count", "nodata", "nan"]
+NALCMS_PROJ4 = (
+    "+proj=laea +lat_0=45 +lon_0=-100 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs=True"
+)
+
+
+class NALCMSZonalStatisticsRasterProcess(Process):
+    """Given files containing vector data and raster data, perform zonal statistics of the overlapping regions"""
+
+    def __init__(self):
+        inputs = [
+            ComplexInput(
+                "shape",
+                "Vector Shape",
+                abstract="An ESRI Shapefile, GML, JSON, GeoJSON, or single layer GeoPackage."
+                " The ESRI Shapefile must be zipped and contain the .shp, .shx, and .dbf.",
+                min_occurs=1,
+                max_occurs=1,
+                supported_formats=[
+                    FORMATS.GEOJSON,
+                    FORMATS.GML,
+                    FORMATS.JSON,
+                    FORMATS.SHP,
+                ],
+            ),
+            ComplexInput(
+                "raster",
+                "Gridded Land Use raster data set",
+                abstract="The Land Use raster to be queried. Default is the CEC NALCMS 2010. Provided raster "
+                "must use the UN FAO Land Cover Classification System (19 types).",
+                metadata=[
+                    Metadata(
+                        "Commission for Environmental Cooperation North American Land Change Monitoring System",
+                        "http://www.cec.org/tools-and-resources/map-files/land-cover-2010-landsat-30m",
+                    ),
+                    Metadata(
+                        "Latifovic, R., Homer, C., Ressl, R., Pouliot, D., Hossain, S.N., Colditz, R.R.,"
+                        "Olthof, I., Giri, C., Victoria, A., (2012). North American land change "
+                        "monitoring system. In: Giri, C., (Ed), Remote Sensing of Land Use and Land "
+                        "Cover: Principles and Applications, CRC-Press, pp. 303-324"
+                    ),
+                ],
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[FORMATS.GEOTIFF],
+            ),
+            LiteralInput(
+                "simple_categories",
+                "Use simplified land classification categories for hydrological "
+                "modeling purposes.",
+                data_type="boolean",
+                default="false",
+                min_occurs=0,
+                max_occurs=1,
+            ),
+            LiteralInput(
+                "band",
+                "Raster band",
+                data_type="integer",
+                default=1,
+                abstract="Band of raster examined to perform zonal statistics.",
+                min_occurs=0,
+                max_occurs=1,
+            ),
```

Comment:
Move identical inputs used across multiple processes to wpsio.py.

Reply:
Will do.

Reply:
Done.